pages/in/register/index.vue

```diff
@@ -41,56 +123,274 @@ export default {
-  },
   methods: {
     ...mapActions([
-      'setAuthDialog',
+      'handleConnectorRedirect',
+      'loginUser',
       'doPostAuthRedirect',
+      'newUser',
+      'updateUserAvatar',
     ]),
-    logPageUnload() {
-      let value = 1;
-      if (window.performance) {
-        value = Math.round(performance.now());
+    async onClickLoginButton() {
+      await this.handleConnectWallet();
+    },
+    onClickDialogCloseButton() {
+      this.currentTab = TAB_OPTIONS.LOGIN;
+      this.errorCode = '';
+      this.error = '';
+      this.signInPayload = {};
+      this.authcoreUserData = {};
+      this.handleConnectWallet();
+    },
+    setError(code, error) {
+      this.currentTab = TAB_OPTIONS.ERROR;
+      this.errorCode = code;
+      this.error = error;
+    },
+
+
+    // Login
+    async handleConnectWallet() {
+      try {
+        await this.connectWallet();
+        const successLogin = await this.login();
+        if (successLogin) {
+          this.redirectAfterSignIn();
+        }
+      } catch (error) {
+      // eslint-disable-next-line no-console
+        this.setError(error.message, error);
+      }
+    },
+    async handleAuthSignIn() {
+      const { method, code } = this.$route.query;
+      if (method && code) {
+        try {
+          const user = await this.handleConnectorRedirect({
+            method,
+            params: { code },
+          });
+          this.authcoreUserData = user;
+          const successLogin = await this.login({ idToken: user.idToken });
+          if (successLogin) {
+            this.redirectAfterSignIn();
+          }
+        } catch (err) {
+          const errData = err.response || err;
+          const errMessage = errData.data || errData.message || errData;
+          this.setError(errMessage, errMessage);
+          console.error(errMessage); // eslint-disable-line no-console
+        }
       }
-      logTimingEvent(this, 'RegFlow', 'CloseRegisterPageTiming', 'CloseRegisterPageTiming', value);
-      logTrackerEvent(this, 'RegFlow', 'CloseRegisterPage', 'CloseRegisterPage', value);
     },
-    logPageload() {
-      let value = 1;
-      if (window.performance) {
-        value = Math.round(performance.now());
+    async login(loginPayload = {}) {
+      let isLoginSuccessful = false;
+      const signer = this.getSigner;
+      const address = this.getAddress;
+      const payload = await signLoginMessage(signer, address);
+      const data = {
+        locale: this.getCurrentLocale,
+        platform: this.platform,
+        ...payload,
+        ...loginPayload,
+      };
+      this.signInPayload = data;
+      try {
+        await this.loginUser(this.signInPayload);
+        isLoginSuccessful = true;
+      } catch (err) {
+        if (err.response) {
+          if (err.response.status === 404) {
+            this.currentTab = TAB_OPTIONS.REGISTER;
+          }
+          isLoginSuccessful = false;
```

Comment:
this is not needed since try block would only throw before setting `isLoginSuccessful`?

Reply:
[♻️ Improve code readability based on comment suggestions](https://github.com/likecoin/like-co/pull/1740/commits/de152633a72f64a26f8041d517b6aef6cbddc2d9)